packages/provider/src/api/captcha.ts

```diff
@@ -167,12 +175,28 @@ export function prosopoRouter(env: ProviderEnvironment): Router {
 		try {
 			const { user, dapp } = GetPowCaptchaChallengeRequestBody.parse(req.body);
 
+			validateAddress(user, false, 42);
+			validateAddress(dapp, false, 42);
```

Comment:
duplicate?

Reply:
No, one is validating the `dapp` and the other is validating the `user`